### DISCUSSION TITLE
Dim inactive regions (C/C++)

### DISCUSSION THREAD

**Discussion by nuid64** [helix-editor/helix#10068] [General]
It would be nice to have highlighted only code which will be compiled. Seen this feature in VSCode cland extension

**Comment by the-mikedavis** [comment]
I assume that clang implements this using diagnostic tags like rust-analyzer does. So this should have been implemented in #9780. You can try using the latest master branch along with a theme that defines `diagnostic.unnecessary`, for example the default theme after that PR was merged: https://github.com/helix-editor/helix/blob/64c1de19884b0eaf43ccb3335115c2cbca8d7432/theme.toml#L83-L84

**Comment by nuid64** [comment]
Oh, that's cool. For some reason this doesn't work with clangd. I can't see it sends any diagnostic tags in helix log

**Comment by pascalkuthe** [comment]
I think clangd does this using semantic highlighting which we don't support (duplicate of #5589).

Could be nice to add support for diagnostic tags upstream instead. For example RA supports semantic highlights yet uses diagnostic tags aswell.
